radiography/src/main/java/radiography/StateRenderer.kt

```diff
@@ -0,0 +1,36 @@
+package radiography
+
+import radiography.StateRenderer.Companion.from
+
+/**
+ * Renders extra attributes for specifics types in the output of [Radiography.scan].
+ * Call [from] to create instances with reified types:
+ * ```
+ *  val myRenderer: StateRenderer<MyView> = from { myView, append ->
+ *    append(myView.customAttributeValue)
+ *  }
+ * ```
+ */
+abstract class StateRenderer<T>(
```

Comment:
Why is this abstract? Just make it final, give it a `@PublishedApi internal` constructor, and make `from` or `stateRendererFor` the only way to create one.

Reply:
Had some trouble with reified + inline and private constructors, your suggestion makes sense.